From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Denis Grigorev <d.grigorev@omprussia.ru>
Date: Fri, 21 May 2021 22:46:28 +0300
Subject: [PATCH 40/67] Adapt build configuration for SailfishOS. JB#53756

Adapts the build configuration to support WebRTC on Sailfish OS..

Signed-off-by: Denis Grigorev <d.grigorev@omprussia.ru>
---
 third_party/libwebrtc/build/config/BUILDCONFIG.gn | 8 +++++++-
 third_party/libwebrtc/build/config/ozone.gni      | 3 ++-
 third_party/libwebrtc/build/config/ui.gni         | 6 +++---
 third_party/libwebrtc/modules/BUILD.gn            | 2 +-
 4 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/third_party/libwebrtc/build/config/BUILDCONFIG.gn b/third_party/libwebrtc/build/config/BUILDCONFIG.gn
index 571ae5f0d60d..cf8677cafaaf 100644
--- a/third_party/libwebrtc/build/config/BUILDCONFIG.gn
+++ b/third_party/libwebrtc/build/config/BUILDCONFIG.gn
@@ -122,6 +122,11 @@ if (current_os == "") {
 # - Don't call exec_script inside declare_args. This will execute the script
 #   even if the value is overridden, which is wasteful. See first bullet.
 
+declare_args() {
+  # Enable special configuration for SFOS
+  is_sfos = current_os == "linux"
+}
+
 declare_args() {
   # Set to enable the official build level of optimization. This has nothing
   # to do with branding, but enables an additional level of optimization above
@@ -135,7 +140,7 @@ declare_args() {
              (current_cpu != "s390x" && current_cpu != "s390" &&
               current_cpu != "ppc64" && current_cpu != "ppc" &&
               current_cpu != "mips" && current_cpu != "mips64" &&
-              current_cpu != "riscv64")
+              current_cpu != "riscv64" && !is_sfos)
 
   # Allows the path to a custom target toolchain to be injected as a single
   # argument, and set as the default toolchain.
@@ -292,6 +297,7 @@ is_bsd = current_os == "openbsd"
 is_mac = current_os == "mac"
 is_nacl = current_os == "nacl"
 is_win = current_os == "win" || current_os == "winuwp"
+is_sfos = is_linux && is_sfos
 
 is_apple = is_ios || is_mac
 is_posix = !is_win && !is_fuchsia
diff --git a/third_party/libwebrtc/build/config/ozone.gni b/third_party/libwebrtc/build/config/ozone.gni
index bf6eb78272c8..a53964dfe460 100644
--- a/third_party/libwebrtc/build/config/ozone.gni
+++ b/third_party/libwebrtc/build/config/ozone.gni
@@ -119,7 +119,8 @@ declare_args() {
   # once all the other places are fixed.
   # TODO(1096425): remove use_x11.
   use_x11 =
-      ozone_platform_x11 && (is_linux || is_bsd) && !is_chromecast && !is_chromeos_lacros
+      ozone_platform_x11 && (is_linux || is_bsd) && !is_chromecast &&
+      !is_chromeos_lacros && !is_sfos
 }
 
 import(ozone_extra_path)
diff --git a/third_party/libwebrtc/build/config/ui.gni b/third_party/libwebrtc/build/config/ui.gni
index 1bcc33e1927c..8a0d15dab180 100644
--- a/third_party/libwebrtc/build/config/ui.gni
+++ b/third_party/libwebrtc/build/config/ui.gni
@@ -32,12 +32,12 @@ declare_args() {
   # True means the UI is built using the "views" framework.
   toolkit_views = is_mac || is_win || is_linux || is_chromeos || is_fuchsia
 
-  use_glib = (is_linux || is_bsd) && !is_chromecast
+  use_glib = (is_linux || is_bsd) && !is_chromecast && !is_sfos
 }
 
-assert(!use_glib || ((is_linux || is_bsd) && !is_chromecast))
+assert(!use_glib || ((is_linux || is_bsd) && !is_chromecast && !is_sfos))
 
-use_atk = (is_linux || is_bsd) && !is_chromecast && use_glib &&
+use_atk = (is_linux || is_bsd) && !is_chromecast && !is_sfos && use_glib &&
           current_toolchain == default_toolchain
 
 # Whether using Xvfb to provide a display server for a test might be
diff --git a/third_party/libwebrtc/modules/BUILD.gn b/third_party/libwebrtc/modules/BUILD.gn
index ffb7491b31d4..76da07af1bb9 100644
--- a/third_party/libwebrtc/modules/BUILD.gn
+++ b/third_party/libwebrtc/modules/BUILD.gn
@@ -23,7 +23,7 @@ group("modules") {
     "video_coding",
   ]
 
-  if (rtc_desktop_capture_supported) {
+  if (rtc_desktop_capture_supported && !is_sfos) {
     deps += [ "desktop_capture" ]
   }
 }
-- 
2.51.0

